mux/tree.go

```diff
@@ -87,6 +87,7 @@ func (t Tree) Match(path string) (Node, middleware.Middleware, context.Params, s
 				if len(orphanMatches) > 0 {
 					for i := 0; i < len(orphanMatches); i++ {
 						m = orphanMatches[i].node.Middleware().Merge(m)
+						m.Reverse()
```

Comment:
First of all adding `Reverse()` adds lots of complexity to a router, slowing it down. Easier is just to change loop above `for i := 0; i < len(orphanMatches); i++`.

Anyway I don't think there is a need to revers order at all. Overall it is working correctly, except cases where static middleware is applied to wildcard route. Given this example:

```go
	router.GET("/x/{param}", func(ctx *fasthttp.RequestCtx) {})

	router.USE(http.MethodGet, "/x/{param}", mockFastHTTPMiddleware("m1"))
	router.USE(http.MethodGet, "/x/y", mockFastHTTPMiddleware("m2"))
```
You would expect m1 to be called first (in order they were created) but because this middleware was added to different tree nodes, and those nodes are ordered, **m2** is called before **m1**.

We could change the order, but it is not a solution. Given the example:
```go
	router.GET("/x/{param}", func(ctx *fasthttp.RequestCtx) {})

	router.USE(http.MethodGet, "/x/{param}", mockFastHTTPMiddleware("m1"))
	router.USE(http.MethodGet, "/x/y", mockFastHTTPMiddleware("m2"))
	router.USE(http.MethodGet, "/x/y", mockFastHTTPMiddleware("m3"))
```

The order here is **m3->m2->m1** which is semi correct, as the order if incorrect only when merging orphan nodes with the main middleware tree. Reversing order for the whole middleware slice, will result in completely different order for all of middleware which is incorrect.

Reply:
> Overall it is working correctly, except cases where static middleware is applied to wildcard route.

The issue for me is that this is my main case of use.

> Reversing order for the whole middleware slice, will result in completely different order for all of middleware which is incorrect.

Totally get your point. Let's elaborate further. I have added a deep nested static case to test if it will break the logic. Please note that the reversing happens only within the oprhaned match loop so it doesn't reorder items on other occasions. So far I couldn't produce incorrect results. 

Benchmarks also remain stellar nevertheless:

```
go test -bench=. -cpu=4
test
goos: linux
goarch: amd64
pkg: github.com/vardius/gorouter/v4
BenchmarkStatic1-4                      58706526                19.7 ns/op
BenchmarkStatic2-4                      45630166                26.5 ns/op
BenchmarkStatic3-4                      33508587                33.4 ns/op
BenchmarkStatic5-4                      21932134                46.0 ns/op
BenchmarkStatic10-4                     12442113                84.7 ns/op
BenchmarkStatic20-4                      7243618               154 ns/op
BenchmarkWildcard1-4                     6618468               181 ns/op
BenchmarkWildcard2-4                     5729289               205 ns/op
BenchmarkWildcard3-4                     5328877               219 ns/op
BenchmarkWildcard5-4                     4603405               255 ns/op
BenchmarkWildcard10-4                    3544700               340 ns/op
BenchmarkWildcard20-4                    2276683               540 ns/op
BenchmarkRegexp1-4                       4256842               273 ns/op
BenchmarkRegexp2-4                       3283544               369 ns/op
BenchmarkRegexp3-4                       2506996               485 ns/op
BenchmarkRegexp5-4                       1734328               655 ns/op
BenchmarkRegexp10-4                      1000000              1219 ns/op
BenchmarkRegexp20-4                       488301              2585 ns/op
BenchmarkFastHTTPStatic1-4              30919155                34.5 ns/op
BenchmarkFastHTTPStatic2-4              25989915                46.4 ns/op
BenchmarkFastHTTPStatic3-4              20417211                60.7 ns/op
BenchmarkFastHTTPStatic5-4              14565698                79.8 ns/op
BenchmarkFastHTTPStatic10-4              8582150               143 ns/op
BenchmarkFastHTTPStatic20-4              5158412               245 ns/op
BenchmarkFastHTTPWildcard1-4            12319296                92.9 ns/op
BenchmarkFastHTTPWildcard2-4             9517089               125 ns/op
BenchmarkFastHTTPWildcard3-4             8331864               146 ns/op
BenchmarkFastHTTPWildcard5-4             6278666               190 ns/op
BenchmarkFastHTTPWildcard10-4            4149974               289 ns/op
BenchmarkFastHTTPWildcard20-4            2362891               506 ns/op
BenchmarkFastHTTPRegexp1-4               6121318               206 ns/op
BenchmarkFastHTTPRegexp2-4               3767931               329 ns/op
BenchmarkFastHTTPRegexp3-4               2734198               447 ns/op
BenchmarkFastHTTPRegexp5-4               1807521               702 ns/op
BenchmarkFastHTTPRegexp10-4               771120              1299 ns/op
BenchmarkFastHTTPRegexp20-4               451314              2562 ns/op
PASS
ok      github.com/vardius/gorouter/v4  52.010s
```